src/classes/OPP_StageToStateMapping_API.cls

```diff
@@ -0,0 +1,149 @@
+*/
+global class OPP_StageToStateMapping_API {
+
+    /******************************************************************************************************
+    * @description Represents the overall state of the Opportunity.
+    * UNCOMMITTED -- Donor has not yet agreed to make the donation.  Donation is unpaid.  This is useful
+    * for soliciting or staging gifts.
+    * COMMITTED --  Donor has agreed to make a donation.  Donation can be unpaid or partially paid.
+    * More money is still expected from the donor.
+    * FINALIZED -- Donation is completely received.  Donation can be fully or partially paid.  No more
+    * money is expected for the donation.
+    */
+    global enum OPPORTUNITY_STATE { UNCOMMITTED, COMMITTED, FINALIZED }
+
+    /******************************************************************************************************
+    * @description Used to Cache Set of Opportunity Stages that mean Opportunity is in a Committed State.
+    */
+    private static Set<String> COMMITTED_STATE_SET = null;
+
+    /******************************************************************************************************
+    * @description Used to Cache Set of Opportunity Stages that mean Opportunity is in a Finalized State.
+    */
+    private static Set<String> FINALIZED_STATE_SET = null;
+
+    /******************************************************************************************************
+    * @description Separator String used to parse Strings into multiple values
+    */
+    private static final String DEFAULT_SEPARATOR_VAL = ';';
+
+    /*******************************************************************************************************
+    * @description Determines the Opportunity State (UNCOMMITTED, COMMITTED, or FINALIZED) from the
+    * Opportunity Stage.  If the Opportunity Stage does not map to a FINALIZED or COMMITTED state,
+    * based on the Household Settings, it will return UNCOMMITTED.  It will always return UNCOMMITTED
+    * if the settings are not configured.  It first checks the Opportunity Stage is a FINALIZED stage.  If
+    * it is, it returns FINALIZED.  If it is not, it checks to see if the Opportunity Stage is a COMMITTED
+    * stage.  If it is, it returns COMMITTED.  Otherwise, it returns UNCOMMITTED.
+    *
+    * NOTE: Opportunity Stage is treated as Case Insensitive, so Closed Won CLOSED WON and closed won are
+    * all the same for this purpose.
+    * @param String opportunityStage The opportunity stage to check the state of
+    * @return OPPORTUNITY_STATE enum value representing the current opportunity state
+    ********************************************************************************************************/
+    global static OPPORTUNITY_STATE determineOppStateFromOppStage(String opportunityStage) {
+
+        initializeStates();
+
+        String oppStageToCheck = String.isNotBlank(opportunityStage) ? opportunityStage.toLowerCase(opportunityStage) : null;
```

Comment:
Just curious, any specific reason that we use toLowerCase(opportunityStage) instead of toLowerCase() here?

Reply:
Nope.  I will correct that.  It was me being redundant.  It was me being redundant.  It was me being redundant.

Reply:
Corrected.